main/src/library/List.flix

```diff
@@ -353,10 +361,10 @@ mod List {
     ///
     pub def map(f: a -> b \ ef, l: List[a]): List[b] \ ef =
         def loop(ll, acc) = match ll {
-            case Nil     => acc
+            case Nil     => reverse(acc)
```

Comment:
I think this breaks the inliner-- because now the tail call has to happen inside the loop. The previous code was better.

Reply:
@jaschdoc What are your thoughts? What happens when loop has to capture reverse like this?

Reply:
I am not sure, but now reverse is free inside `loop` which might mess with inliner/optimizer/lambda lift.